apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/asyncprofiler/AsyncProfilerDataSender.java

```diff
@@ -0,0 +1,194 @@
+                            buf.flip();
+                            AsyncProfilerData asyncProfilerData = AsyncProfilerData.newBuilder()
+                                    .setContent(ByteString.copyFrom(buf))
+                                    .build();
+                            requestStream.onNext(asyncProfilerData);
+                            buf.clear();
+                        }
+                    } catch (IOException e) {
+                        LOGGER.error("Failed to read JFR file and failed to upload to oap", e);
+                    }
+                }
+
+                requestStream.onCompleted();
+            }
+
+            @Override
+            public void onError(Throwable t) {
+                status.finished();
+                LOGGER.error(t, "Send async profiler task data to collector fail with a grpc internal exception.");
+                ServiceManager.INSTANCE.findService(GRPCChannelManager.class).reportError(t);
+            }
+
+            @Override
+            public void onCompleted() {
+                status.finished();
+            }
+        });
+        AsyncProfilerMetaData metaData = AsyncProfilerMetaData.newBuilder()
+                .setService(Config.Agent.SERVICE_NAME)
+                .setServiceInstance(Config.Agent.INSTANCE_NAME)
+                .setType(AsyncProfilingStatus.PROFILING_SUCCESS)
+                .setContentSize(size)
+                .setTaskId(task.getTaskId())
+                .build();
+        AsyncProfilerData asyncProfilerData = AsyncProfilerData.newBuilder().setMetaData(metaData).build();
+        dataStreamObserver.onNext(asyncProfilerData);
+
+        status.wait4Finish();
+    }
+
+    public void sendError(AsyncProfilerTask task, String errorMessage) {
+        if (status != GRPCChannelStatus.CONNECTED) {
+            return;
+        }
+        final GRPCStreamServiceStatus status = new GRPCStreamServiceStatus(false);
+        StreamObserver<AsyncProfilerData> dataStreamObserver = asyncProfilerTaskStub.withDeadlineAfter(
+                GRPC_UPSTREAM_TIMEOUT, TimeUnit.SECONDS
+        ).collect(new StreamObserver<AsyncProfilerCollectionResponse>() {
+            @Override
+            public void onNext(AsyncProfilerCollectionResponse value) {
+            }
+
+            @Override
+            public void onError(Throwable t) {
+                status.finished();
+                ServiceManager.INSTANCE.findService(GRPCChannelManager.class).reportError(t);
+            }
+
+            @Override
+            public void onCompleted() {
+                status.finished();
+            }
+        });
+        AsyncProfilerMetaData metaData = AsyncProfilerMetaData.newBuilder()
+                .setService(Config.Agent.SERVICE_NAME)
+                .setServiceInstance(Config.Agent.INSTANCE_NAME)
+                .setTaskId(task.getTaskId())
+                .setType(AsyncProfilingStatus.EXECUTION_TASK_ERROR)
+                .setContentSize(0)
```

Comment:
```suggestion
                .setContentSize(-1)
```

Usually negative value indicates null for length.